resourcex/traverse.go

```diff
@@ -73,3 +73,47 @@ func Traverse(v resource.PropertyValue, p resource.PropertyPath, f func(resource
 		}
 	}
 }
+
+// Walk traverses a property value along all paths, performing a depth first search.
+func Walk(v resource.PropertyValue, f func(resource.PropertyValue, WalkState)) {
+	walk(v, f, WalkState{})
+}
+
+type WalkState struct {
+	// We are entering the value. The children of the value have not yet been visited.
+	Entering bool
+	// The value is not transitively computed.
```

Comment:
This is a bit weird and does not mesh with my mental model:

```
type Value = Unknown
                    | Known ConcreteValue
```

Are there really transitively unknown values or this is just a degenerate encoding artifact that does not occur in practice?

Reply:
I'm guessing it's a degenerate value, but lots of serde stuff requires valid types, even on computed fields so it might come up. This is mostly to be cautious.